### PR TITLE
docs: update readme to add allowlist formats

### PR DESCRIPTION
**Pull request by andrewdetorres**
PR to add allowlist formats to the README.md to help make the options available to a user more clear

Would appreciate any feedback on this PR. 😄 

In reference to https://github.com/IBM/audit-ci/issues/252

Thanks,
Andrew

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,39 @@ Also, it suppresses an advisory of `axios` and a transitive advisory of `react-s
 }
 ```
 
+### Allowlist formats
+
+To suppress the vulnerability associated with the an advisory. You can simply add the advisory ID to the allowlist.
```
Comment (quinnturner) [inline README.md:70]: This section is a great start. It leans a bit too much towards examples compared to definitions for my liking. There's an example section below that captures most of the examples you've described here. Here's my attempt, let me know what you think:

### Allowlisting

Allowlists are a mechanism to suppress an advisory warning from the audit. A team may want to suppress an advisory when:

- A fix has already been started
- There is no bandwidth to fix the advisory
- The risk is tolerable for the project
- The advisory is inaccurate or incorrect
- The vulnerable code is not actually used

An allowlist may contain multiple allowlist records. The allowlist record format can be written into one of three categories:

- `module` allowlist record (e.g., `axios`, suppressing all advisories _directly_ caused by `axios`, **not transitive advisories**)
- `advisory` allowlist record (e.g. `GHSA-42xw-2xvc-qx8m`, suppresses all instances of advisory based on the GitHub advisory identifier)
- `path` allowlist record (e.g. `GHSA-rp65-9cf3-cjxr|react-scripts>@svgr/webpack>@svgr/plugin-svgo>svgo>css-select>nth-check`, the specific and full advisory path **with wildcard support**)

When `audit-ci` identifies new advisories at or above the configured level, the CI pipeline will fail.

```txt
Found vulnerable advisory paths:
GHSA-pw2r-vq6v-hr8c|axios>follow-redirects
GHSA-74fj-2j2h-c42q|axios>follow-redirects
GHSA-4w2v-q235-vp99|axios
GHSA-42xw-2xvc-qx8m|axios
GHSA-cph5-m8f7-6c5x|axios
Failed security audit due to high, moderate vulnerabilities.
Vulnerable advisories are:
https://github.com/advisories/GHSA-pw2r-vq6v-hr8c
https://github.com/advisories/GHSA-74fj-2j2h-c42q
https://github.com/advisories/GHSA-4w2v-q235-vp99
https://github.com/advisories/GHSA-42xw-2xvc-qx8m
https://github.com/advisories/GHSA-cph5-m8f7-6c5x
Exiting...
```

In this case, the advisories can be wholly suppressed in several different ways.

First, the most granular and secure approach, using full paths. If in the future, the same advisory arises with a different path, the pipeline will fail.

```jsonc
"allowlist": [
  "GHSA-pw2r-vq6v-hr8c|axios>follow-redirects",
  "GHSA-74fj-2j2h-c42q|axios>follow-redirects",
  "GHSA-4w2v-q235-vp99|axios",
  "GHSA-42xw-2xvc-qx8m|axios",
  "GHSA-cph5-m8f7-6c5x|axios"
]
```

The next best approach would be to suppress the advisories using the advisory ID. This approach can be useful if your team knows that the application is not (and will not be) affected by the advisory regardless of the path. Often, the same advisory can be present in many paths. Allowlisting by advisory ID is terser than the alternative of listing all paths.

```jsonc
"allowlist": [
  "GHSA-pw2r-vq6v-hr8c",
  "GHSA-74fj-2j2h-c42q",
  "GHSA-4w2v-q235-vp99",
  "GHSA-42xw-2xvc-qx8m",
  "GHSA-cph5-m8f7-6c5x"
]
```

The next approach is to allowlist the modules themselves. All current and future advisories are suppressed automatically when using module allowlist records. Compared to other suppression approaches, there's an increased risk of a new advisory impacting your application due to the broad suppression. Suppressing via a module allowlist record is often less useful than using path allowlist records + wildcards, as noted in the final approach.

```jsonc
"allowlist": [
  "axios",
  "follow-redirects"
]
```

Finally, wildcards can be used within path allowlist records. Wildcards are useful for trusted development-only dependencies such as `react-scripts`. Unlike the module allowlist record of `react-scripts`, the path allowlist of `*|react-scripts>*` suppresses transitive dependency advisories (dependencies of dependencies).

Wildcard matching works by:

1. splitting the allowlist record at every wildcard
2. constructing a regex matching anything at each wildcard location

An allowlist record may include any number of wildcards such as `*|react-scripts>*>*>example>*`

**Comment by quinnturner** [comment]
I've opened up https://github.com/IBM/audit-ci/pull/254 as an alternative to this PR. It contains reasoning for when and why each approach should be used (perhaps a bit too much). I tried to ensure that what you put here was captured in some combo of my docs and the examples. I am hoping you can review my PR @andrewdetorres!

**Comment by andrewdetorres** [comment]
Looks great. Reviewed and approved new PR 😅

**Comment by quinnturner** [comment]
Thanks again @andrewdetorres :)

**Pull request closed without merging**